src/shader.cpp

```diff
@@ -233,29 +244,37 @@ bool ShaderProgram::setShaderFromSourceFile(QString fileName)
 {
     QFile file(fileName);
     if (!file.open(QFile::ReadOnly))
+    {
+        //g_logger.error("Could not read Shader:\n%s", fileName.toStdString());
         return false;
+    }
     return setShader(file.readAll());
 }
 
 
 bool ShaderProgram::setShader(QString src)
 {
-    std::unique_ptr<Shader> vertexShader(new Shader(QGLShader::Fragment, m_context));
-    std::unique_ptr<Shader> fragmentShader(new Shader(QGLShader::Vertex, m_context));
+    std::unique_ptr<Shader> vertexShader(new Shader(QGLShader::Fragment));
+    std::unique_ptr<Shader> fragmentShader(new Shader(QGLShader::Vertex));
     //tfm::printf("Shader source:\n###\n%s\n###\n", src.toStdString());
-    if(!vertexShader->compileSourceCode(src.toAscii()))
+#ifdef DISPLAZ_USE_QT4
+    QByteArray src_ba = src.toAscii();
+#else
+    QByteArray src_ba = src.toLatin1();
+#endif
```

Comment:
Well, it's unclear whether shader compilers in typical drivers allow utf-8, but I suggest we assume they do until we see this causing problems in the wild.  The following link suggests utf-8 source encoding will be part of the future spec anyway:

https://www.khronos.org/bugzilla/show_bug.cgi?id=760



Ideally we'd just strip out any weird characters in comments, but `toAscii()` never really achieved that anyway.